pyjobshop/solvers/ortools/Constraints.py

```diff
@@ -200,13 +200,12 @@ def _previous_before_constraints(self):
                     if Constraint.BEFORE in sequencing_constraints:
                         seq_var.activate(model)
 
-                        idx1 = seq_var.mode_vars.index(var1)
-                        idx2 = seq_var.mode_vars.index(var2)
-                        rank1 = seq_var.ranks[idx1]
-                        rank2 = seq_var.ranks[idx2]
+                        start1 = var1.start
+                        start2 = var2.start
                         both_present = [var1.is_present, var2.is_present]
+                        expr = start1 <= start2
 
-                        model.add(rank1 <= rank2).only_enforce_if(both_present)
+                        model.add(expr).only_enforce_if(both_present)
```

Comment:
I will give it a try. Towards that end:

- The setup of `_previous_before_constraints` can be simplified now I guess. Is that something we want? 
- Do we want to rename `Constraint.PREVIOUS` now to the shorter `Constraint.BEFORE` :-)?

Reply:
- Yes we can simplify that too
- I think BACK_TO_BACK is most suitable. It directly conveys to the user what the constraint is about. What do you think?

Reply:
Yes, perhaps BACK_TO_BACK is more saying than BEFORE or something like CONSECUTIVE.

Reply:
I like CONSECUTIVE!

Reply:
I like CONSECUTIVE!

Reply:
OK, I will use that. Saves also a letter 😀

Reply:
While working on this, I wondered what the use case is of PREVIOUS (or CONSECUTIVE)? In particular, it ensures that two tasks are scheduled consecutively when scheduled on the same resource. So apparently, when they are scheduled on different resource, there is no restriction/link between them. I have a hard time thinking of a use case of this. 

I can imagine you want two tasks always be consecutively scheduled. In that case, I think it would be best to just make one ("umbrella") task of those consecutive tasks.

Reply:
In the animal feed application, consecutively scheduling tasks on the press lines is always paired with the SAME_RESOURCE constraint. There's also another paper I recall that calls this "contiguity" (same resource and consecutive) but I borrowed previous from CP Optimizer. I can also not think of a situation where you want to consecutively schedule without using the same resource.

> I can imagine you want two tasks always be consecutively scheduled. In that case, I think it would be best to just make one ("umbrella") task of those consecutive tasks.

This does not work if you have waiting times between two consecutive tasks, e.g., the press line example when the GML is the bottleneck.

Reply:
Furthermore, I noticed in `ortools.Constraints._consecutive_constraints()`:
```
if not isinstance(data.resources[resource], Machine):
    continue  # skip sequencing on non-machine resources
```
So it seems that it does not apply to resources in general. However, in `Model` we have the following description:
```
def add_consecutive(self, task1: Task, task2: Task):
        """
        Adds a constraint that the first task must be scheduled right before
        the second task, meaning that no task is allowed to schedule between,
        on resources that they are both scheduled on.
        """
        self._add_constaint(task1, task2, Constraint.CONSECUTIVE)
```
Shouldn't resources here be machines?